File: clientPython/zeroc-ice-3.7.2/java/test/src/main/java/test/Ice/dispatcher/TestI.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

package test.Ice.dispatcher;

import test.Ice.dispatcher.Test.TestIntf;

public class TestI implements TestIntf
{
    private static void test(boolean b)
    {
        if(!b)
        {
            throw new RuntimeException();
        }
    }

    TestI(Dispatcher dispatcher)
    {
        _dispatcher = dispatcher;
    }

    @Override
    public void op(com.zeroc.Ice.Current current)
    {
        test(_dispatcher.isDispatcherThread());
    }

    @Override
    public void
    sleep(int to, com.zeroc.Ice.Current current)
    {
        try
        {
            Thread.sleep(to);
        }
        catch(InterruptedException ex)
        {
            System.err.println("sleep interrupted");
        }
    }

    @Override
    public void opWithPayload(byte[] seq, com.zeroc.Ice.Current current)
    {
        test(_dispatcher.isDispatcherThread());
    }

    @Override
    public void shutdown(com.zeroc.Ice.Current current)
    {
        current.adapter.getCommunicator().shutdown();
    }

    private Dispatcher _dispatcher;
}

File: clientPython/zeroc-ice-3.7.2/python/test/Ice/dispatcher/TestI.py
#
# Copyright (c) ZeroC, Inc. All rights reserved.
#

import Ice, Test, Dispatcher, time

def test(b):
    if not b:
        raise RuntimeError('test assertion failed')

class TestIntfI(Test.TestIntf):
    def op(self, current=None):
        test(Dispatcher.Dispatcher.isDispatcherThread())

    def sleep(self, ms, current=None):
        time.sleep(ms / 1000.0)

    def opWithPayload(self, bytes, current=None):
        test(Dispatcher.Dispatcher.isDispatcherThread())

    def shutdown(self, current=None):
        test(Dispatcher.Dispatcher.isDispatcherThread())
        current.adapter.getCommunicator().shutdown()

class TestIntfControllerI(Test.TestIntfController):
    def __init__(self, adapter):
        self._adapter = adapter

    def holdAdapter(self, current=None):
        test(Dispatcher.Dispatcher.isDispatcherThread())
        self._adapter.hold()

    def resumeAdapter(self, current=None):
        test(Dispatcher.Dispatcher.isDispatcherThread())
        self._adapter.activate()
